File: week6/Week6Day4/app/src/main/java/com/example/luisenriquez/week6day4/data/remote/RemoteServiceHelper.java
package com.example.luisenriquez.week6day4.data.remote;

import com.example.luisenriquez.week6day4.data.remote.models.MovieResponse;
import com.example.luisenriquez.week6day4.utils.Constants;
import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;

import java.util.List;

import io.reactivex.Single;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RemoteServiceHelper
{
    public Single<MovieResponse> getMovieResponse(String srchItm)
    {
        Retrofit myRetro = getRetroFit(Constants.MOVIE_API_BASE_URL);
        RemoteService myService = myRetro.create(RemoteService.class);

        return myService.getMovieResponse(Constants.MOVIE_API_KEY,
                                            srchItm);
    }

    private Retrofit getRetroFit(String baseUrl)
    {
        HttpLoggingInterceptor MyInterceptor = new HttpLoggingInterceptor();

        MyInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient MyClient = new OkHttpClient.Builder()
                .addInterceptor(MyInterceptor)
                .build();

        return new Retrofit.Builder()
                .baseUrl(baseUrl)
                .client(MyClient)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();
    }
}

File: Week7/Week7Day4Test/app/src/main/java/com/example/luisenriquez/week7day4test/data/remote/RemoteServiceHelper.kt
package com.example.luisenriquez.week7day4test.data.remote

import com.example.luisenriquez.week7day4test.data.remote.model.NumbersResponse
import com.example.luisenriquez.week7day4test.utils.Constants
import com.jakewharton.retrofit2.adapter.kotlin.coroutines.experimental.CoroutineCallAdapterFactory
import kotlinx.coroutines.experimental.Deferred
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

class RemoteServiceHelper
{
    fun getRandomNumbers() : Deferred<NumbersResponse>
    {
        val retrofit = getRetrofit(Constants.BASE_URL)
        val service = retrofit.create(RemoteService::class.java)

        return service.getRandomNumbers()
    }

    private fun getRetrofit (baseUrl : String) : Retrofit
    {
        val interceptor = HttpLoggingInterceptor()

        interceptor.level = HttpLoggingInterceptor.Level.BODY

        val okHttpClient = OkHttpClient.Builder()
                                        .addInterceptor(interceptor)
                                        .build()

        return Retrofit.Builder()
                        .baseUrl(baseUrl)
                        .client(okHttpClient)
                        .addConverterFactory(GsonConverterFactory.create())
                        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                        .addCallAdapterFactory(CoroutineCallAdapterFactory())
                        .build()
    }


}